portfolio/src/main/java/com/google/sps/servlets/FormServlet.java

```diff
@@ -50,20 +50,37 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Retrieve the maximum quantity of comments to display
+    String radioValueString = request.getParameter("value");
+    int max = Integer.parseInt(radioValueString);
+
+    // Obtain a list of at most 'max' comments
     Query query = new Query("Comment");
     PreparedQuery results = datastore.prepare(query);
```

Comment:
Query can set the limit for you, upon construction you can do more sophisticated logic including setting the limit on results. So you don't need to write this logic yourself.

Limit:
https://cloud.google.com/datastore/docs/concepts/queries#cursors_limits_and_offsets

Reply:
I explored the documentation you linked, but when I tried using snippets of the code they were not recognized (the query constructor is different)--the dependency I'm using is Google App Engine & I believe that dependency is for Cloud Datastore. I explored the App Engine Query documentation (https://cloud.google.com/appengine/docs/standard/java/javadoc/com/google/appengine/api/datastore/Query) but did not seem to find anything similar to setLimit. 

Should I leave the manual check or try to explore other means of setting the limit?

Reply:
If you look on the original link I gave, they use `Query.newEntityQueryBuilder()` builder pattern

It looks like it actually is a different library, but you can set it with your builder with `QueryOptions`

https://cloud.google.com/appengine/docs/standard/java/javadoc/com/google/appengine/api/search/Query.Builder#setoptions_1

https://cloud.google.com/appengine/docs/standard/java/javadoc/com/google/appengine/api/search/QueryOptions.html

Reply:
I finally got it working! I tried to use Query-related settings for awhile, but was having many issues and then decided to search for other documentation. I found this regarding Fetch-related settings:
https://cloud.google.com/appengine/docs/standard/java/javadoc/com/google/appengine/api/datastore/FetchOptions.html
It took awhile playing around with types and looking at examples online, but after reading I feel like I better understand how to manipulate some of the fetch parameters.